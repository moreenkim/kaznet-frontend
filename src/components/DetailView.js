import React, { Component } from 'react';
import { Table, Col } from 'reactstrap';

export default class DetailView extends Component {
  render() {
    return (
      <Col sm="12" md={{ size: 8, offset: 2 }}>
        <Table borderless className="kaznet-table details">
          <tbody>
            {this.props.renderMainDetails}
            <tr>
              <td className="kaznet-title-h1">Details</td>
            </tr>
            {this.props.renderAdditionalDetails}
          </tbody>
        </Table>
      </Col>
    );
  }
}
