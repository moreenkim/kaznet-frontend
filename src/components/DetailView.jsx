import React from 'react';
import { Table, Col } from 'reactstrap';
import PropTypes from 'prop-types';

const DetailView = props => {
  const { renderMainDetails, renderAdditionalDetails } = props;

  return (
    <Col sm="12" md={{ size: 8, offset: 2 }}>
      <Table borderless className="kaznet-table details">
        <tbody>
          {renderMainDetails}
          <tr>
            <td className="kaznet-title-h1">Details</td>
          </tr>
          {renderAdditionalDetails}
        </tbody>
      </Table>
    </Col>
  );
};

DetailView.propTypes = {
  renderMainDetails: PropTypes.shape({}).isRequired,
  renderAdditionalDetails: PropTypes.shape({}).isRequired
};

export default DetailView;
