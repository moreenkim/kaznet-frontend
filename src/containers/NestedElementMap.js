import React, { Component } from 'react';
import ElementMap from './ElementMap';

export default class NestedElementMap extends Component {
  render() {
    const { detailitems } = this.props;
    const details = [];
    let counter = 1;

    for (const [key, value] of Object.entries(detailitems)) {
      details.push(
        React.createElement('tr', { key }, [
          <ElementMap
            key={counter}
            items={[key]}
            HTMLTag={this.props.HTMLTag}
            className="kaznet-title-td"
          />,
          // Need the key to be different as such we are multiplying it by 20 from the first one.
          <ElementMap key={counter * 20} items={[value]} HTMLTag={this.props.HTMLTag} />
        ])
      );
      counter++;
    }

    return details;
  }
}
