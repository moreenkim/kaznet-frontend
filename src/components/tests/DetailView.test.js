// Test DetailView
import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

import DetailView from "../DetailView";
import NestedElementMap from "../../containers/NestedElementMap";
import ErrorBoundary from "../ErrorBoundary";

describe("components/DetailView", () => {
  it("renders without crashing", () => {
    const renderMainDetails = function() {
      const detailItems = {
        Name: "Dave",
        Test: "Testing"
      };
      return <NestedElementMap detailitems={detailItems} HTMLTag="td" />;
    };

    const renderAdditionalDetails = function() {
      const detailItems = {
        Gender: "Male"
      };
      return <NestedElementMap detailitems={detailItems} HTMLTag="td" />;
    };

    shallow(
      <DetailView
        renderMainDetails={renderMainDetails}
        renderAdditionalDetails={renderAdditionalDetails}
      />
    );
  });

  it("renders a view", () => {
    const renderMainDetails = function() {
      const detailItems = {
        Name: "Dave",
        Test: "Testing"
      };
      return <NestedElementMap detailitems={detailItems} HTMLTag="td" />;
    };

    const renderAdditionalDetails = function() {
      const detailItems = {
        Gender: "Male"
      };
      return <NestedElementMap detailitems={detailItems} HTMLTag="td" />;
    };

    const wrapper = mount(
      <DetailView
        renderMainDetails={renderMainDetails()}
        renderAdditionalDetails={renderAdditionalDetails()}
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
