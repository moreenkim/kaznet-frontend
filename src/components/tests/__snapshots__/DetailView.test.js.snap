// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/DetailView renders a view 1`] = `
<DetailView
  renderAdditionalDetails={
    <NestedElementMap
      HTMLTag="td"
      detailitems={
        Object {
          "Gender": "Male",
        }
      }
    />
  }
  renderMainDetails={
    <NestedElementMap
      HTMLTag="td"
      detailitems={
        Object {
          "Name": "Dave",
          "Test": "Testing",
        }
      }
    />
  }
>
  <Col
    md={
      Object {
        "offset": 1,
        "size": 10,
      }
    }
    sm="12"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <div
      className="col-sm-12 col-md-10 offset-md-1"
    >
      <Table
        borderless={true}
        className="kaznet-table details"
        responsiveTag="div"
        tag="table"
      >
        <table
          className="kaznet-table details table table-borderless"
        >
          <tbody>
            <NestedElementMap
              HTMLTag="td"
              detailitems={
                Object {
                  "Name": "Dave",
                  "Test": "Testing",
                }
              }
            >
              <tr
                key="Name"
              >
                <ElementMap
                  HTMLTag="td"
                  className="kaznet-title-td"
                  items={
                    Array [
                      "Name",
                    ]
                  }
                  key="1"
                >
                  <td
                    className="kaznet-title-td"
                    key="0"
                  >
                    Name
                  </td>
                </ElementMap>
                <ElementMap
                  HTMLTag="td"
                  items={
                    Array [
                      "Dave",
                    ]
                  }
                  key="20"
                >
                  <td
                    key="0"
                  >
                    Dave
                  </td>
                </ElementMap>
              </tr>
              <tr
                key="Test"
              >
                <ElementMap
                  HTMLTag="td"
                  className="kaznet-title-td"
                  items={
                    Array [
                      "Test",
                    ]
                  }
                  key="2"
                >
                  <td
                    className="kaznet-title-td"
                    key="0"
                  >
                    Test
                  </td>
                </ElementMap>
                <ElementMap
                  HTMLTag="td"
                  items={
                    Array [
                      "Testing",
                    ]
                  }
                  key="40"
                >
                  <td
                    key="0"
                  >
                    Testing
                  </td>
                </ElementMap>
              </tr>
            </NestedElementMap>
            <tr>
              <td
                className="kaznet-title-h1"
              >
                Details
              </td>
            </tr>
            <NestedElementMap
              HTMLTag="td"
              detailitems={
                Object {
                  "Gender": "Male",
                }
              }
            >
              <tr
                key="Gender"
              >
                <ElementMap
                  HTMLTag="td"
                  className="kaznet-title-td"
                  items={
                    Array [
                      "Gender",
                    ]
                  }
                  key="1"
                >
                  <td
                    className="kaznet-title-td"
                    key="0"
                  >
                    Gender
                  </td>
                </ElementMap>
                <ElementMap
                  HTMLTag="td"
                  items={
                    Array [
                      "Male",
                    ]
                  }
                  key="20"
                >
                  <td
                    key="0"
                  >
                    Male
                  </td>
                </ElementMap>
              </tr>
            </NestedElementMap>
          </tbody>
        </table>
      </Table>
    </div>
  </Col>
</DetailView>
`;
